### DISCUSSION TITLE
[binarycaching] Provide an option to use an alternative to zip on *NIX systems

### DISCUSSION THREAD

**Discussion by davidchisnall** [microsoft/vcpkg#19373] [Ideas]
**Is your feature request related to a problem? Please describe.**

`zip` is not part of POSIX and so is not included in the base install of most *NIX systems.  `pax` is mandated by POSIX and `tar` is typically installed as a non-standard extension.  There's a message interleaved with other output while building that zip isn't installed and so caches can't be used.

**Proposed solution**

Provide a mechanism for platforms to specify their preferred archive / restore tool.  For example, on FreeBSD the preferred tool would likely be `tar --lz4` (in the base system, significantly faster than `zip`).  

Note that GitHub can generate tarballs via the same API that it generates zip files and so there is no requirement for even `unzip` for ports that come from GitHub.  For ports from other sources, it is common to provide tarballs rather than zip files for the sources for building on *NIX.

**Describe alternatives you've considered**

I can install `zip`, but it is not a great user experience to discover that the lack of a non-standard package has made things slower.

**Comment by JackBoosY** [comment]
I think we can accept this.
